packages/orchestration/src/exos/chain-account-kit.js

```diff
@@ -125,49 +125,60 @@ export const prepareChainAccountKit = (zone, { watch, when }) =>
           return this.state.port;
         },
         executeTx() {
-          throw new Error('not yet implemented');
+          return asVow(() => {
+            throw new Error('not yet implemented');
+          });
         },
         /**
          * Submit a transaction on behalf of the remote account for execution on
          * the remote chain.
          *
          * @param {AnyJson[]} msgs
          * @param {Omit<TxBody, 'messages'>} [opts]
-         * @returns {Promise<string>} - base64 encoded bytes string. Can be
-         *   decoded using the corresponding `Msg*Response` object.
+         * @returns {Vow<string>} - base64 encoded bytes string. Can be decoded
+         *   using the corresponding `Msg*Response` object.
          * @throws {Error} if packet fails to send or an error is returned
          */
-        async executeEncodedTx(msgs, opts) {
-          const { connection } = this.state;
-          // TODO #9281 do not throw synchronously when returning a promise; return a rejected Vow
-          /// see https://github.com/Agoric/agoric-sdk/pull/9454#discussion_r1626898694
-          if (!connection) throw Fail`connection not available`;
-          return when(
-            watch(
+        executeEncodedTx(msgs, opts) {
+          return asVow(() => {
+            const { connection } = this.state;
+            // TODO #9281 do not throw synchronously when returning a promise; return a rejected Vow
+            /// see https://github.com/Agoric/agoric-sdk/pull/9454#discussion_r1626898694
+            if (!connection) throw Fail`connection not available`;
```

Comment:
I'm partial to doing that on one line, but I'm pretty sure others prefer...

```suggestion
            if (!connection) {
              throw Fail`connection not available`;
            }
```

I'm not sure why our lint tools don't require that. Maybe it's OK.

Reply:
```
if (!connection) throw Fail`connection not available`;
```
doesn't seem too different from:
```
connection || Fail`connection not available`;
```
from a readability standpoint.

The latter I understand to be a preferred pattern, but it unfortunately does not satisfy `ts-check`.

I've implemented your other suggestions, thanks for the review!